tests/schema/test_custom_fields.py

```diff
@@ -0,0 +1,177 @@
+import textwrap
+from typing import Any, Dict, List, Optional, cast
+
+import strawberry
+from strawberry import StrawberryField
+
+
+def test_simple_custom_field():
+    class UpperCaseField(StrawberryField):
+        def get_result(self, source: Any, info: Any, arguments: Dict[str, Any]) -> str:
+            result = super().get_result(source, info, arguments)
+            return cast(str, result).upper()
+
+    @strawberry.type
+    class Query:
+        name: str = UpperCaseField(default="Patrick")
+
+        @UpperCaseField()
+        def alt_name() -> str:
+            return "patrick91"
+
+    schema = strawberry.Schema(query=Query)
+
+    result = schema.execute_sync("{ name, altName }", root_value=Query())
+
+    assert not result.errors
+    assert result.data["name"] == "PATRICK"
+    assert result.data["altName"] == "PATRICK91"
+
+
+def test_modify_arguments():
+    class PageNumberPagination(StrawberryField):
+        def get_result(
+            self, source: Any, info: Any, arguments: Dict[str, Any]
+        ) -> List[Any]:
+            first = arguments.pop("first", None)
+            result = super().get_result(source, info, arguments)
+
+            # We're going to assume that the resolver returns the right thing
+            # here
+            result = cast(List[Any], result)
+
+            if first is not None:
+                return result[:first]
+            return result
+
+        def get_arguments(self) -> Dict[str, object]:
+            arguments = super().get_arguments()
+            arguments["first"] = Optional[int]
+
+            return arguments
+
+    @strawberry.type
+    class Query:
+        @PageNumberPagination()
```

Comment:
Does this need to be instantiated?

Reply:
It's just simpler, otherwise we would have to do the awkward dance of checking which way the decorator is called.